app/controllers/katello/api/v2/alternate_content_sources_controller.rb

```diff
@@ -134,6 +131,20 @@ def acs_params
       params.require(:alternate_content_source).permit(*keys).to_h.with_indifferent_access
     end
 
+    def check_params_for_invalid_updates
+      # Check parameters which cannot be validated at the model level, throwing
+      # errors where neccessary
```

Comment:
Why can't these be validated at the model level?

Reply:
I think the example you brought up could work very well here (https://github.com/Katello/katello/blob/master/app/models/katello/provider.rb#L48-L56).  If that does indeed work, it would be much nicer to get all of this out of the controller!

Reply:
Thank you for the suggestion + resources. Implemented the necessary changes.